docs/reference-use-cases.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Cluster API Reference Use Cases
+creation-date: 2019-04-16
+last-updated: 2019-04-16
+---
+
+# Cluster API Reference Use Cases
```

Comment:
@vincepri how about adding some content for Machine health check as well like 
- https://github.com/openshift/machine-api-operator#crds
- https://github.com/openshift/machine-api-operator/blob/master/pkg/apis/healthchecking/v1alpha1/machinehealthcheck_types.go

Reply:
Would you be willing to open a PR after this is merged to add new use cases?

Reply:
@vincepri Yes, I will follow up for a PR after this merged.